plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcIdentityCacheMapping.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+public interface JdbcIdentityCacheMapping
+{
+    JdbcIdentityCacheKey getRemoteUserCacheKey(JdbcIdentity identity);
+
+    interface JdbcIdentityCacheKey {}
```

Comment:
This would need to define contract for equals, hashCode, thread safety.

However, it can be pretty limiting to have cache key being opaque extension point.
At the very least, it becomes very hard to reason about correctness of the system.
I can imagine designs where this is the way to go (API-centric integrations like OSGi),
but I do really hope we can replace all this with a boolean toggle, provided by the connector
implementation.